pcsx2/DEV9/ATA/ATA_Info.cpp

```diff
@@ -309,10 +312,13 @@ void ATA::CreateHDDinfo(u64 sizeSectors)
 	//Streaming Performance Granularity
 	//98-99
 	//Total Number of User Addressable Sectors for the 48-bit Address feature set.
-	index = 100 * 2;
-	WriteUInt64(identifyData, &index, static_cast<u16>(nbSectors));
-	index -= 2;
-	WriteUInt16(identifyData, &index, 0); //truncate to 48bits
+	if (lba48Supported)
+	{
+		index = 100 * 2;
+		WriteUInt64(identifyData, &index, nbSectors);
+		index -= 2;
+		WriteUInt16(identifyData, &index, 0); //truncate to 48bits
+	}
```

Comment:
```suggestion
	index = 100 * 2;
	WriteUInt64(identifyData, &index, lba48Supported ? nbSectors : 0);
```

We should limit this to 0000FFFFFFFFFFFFh, however, limiting the value to 32bits sizes above would already cover that.
With the limit, the truncate write is redundant.

Reply:
yes, but ps2 supports 48-bit so the code is valid